src/main/java/com/project/controller/ApplicationController.java

```diff
@@ -0,0 +1,72 @@
+package com.project.controller;
+
+import com.project.exception.ApplicationNotFoundException;
+import com.project.exception.UserException;
+import com.project.model.Application;
+import com.project.model.response.ApplicationResponse;
+import com.project.service.ApplicationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("api/applications")
+public class ApplicationController {
+    private final ApplicationService applicationService;
+
+    @Autowired
+    public ApplicationController(ApplicationService applicationService) {
+        this.applicationService = applicationService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ApplicationResponse>> fetchApplications() {
+        List<Application> applications = applicationService.getAll();
+        List<ApplicationResponse> response = new ArrayList<>();
+
+        for (Application app : applications) {
```

Comment:
Here you could use Java 8 map to simpligy and optimize your code.
https://mkyong.com/java8/java-8-streams-map-examples/

Reply:
replaced for each loops with java 8 stream map